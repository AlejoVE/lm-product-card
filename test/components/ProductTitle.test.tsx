import React from 'react';

import renderer from 'react-test-renderer';
import { ProductTitle, ProductCard } from '../../src';
import { product } from '../data/products';

describe('ProductTitle', () => {
  test('should show title correctly with custom title', () => {
    const wrapper = renderer.create(<ProductTitle title="My custom title" />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('should show component with the name of the product', () => {
    const wrapper = renderer.create(
      <ProductCard product={product}>{() => <ProductTitle />}</ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
