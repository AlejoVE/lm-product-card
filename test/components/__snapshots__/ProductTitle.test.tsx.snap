// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle should show component with the name of the product 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug - card
  </span>
</div>
`;

exports[`ProductTitle should show title correctly with custom title 1`] = `
<span
  className="productDescription undefined"
>
  My custom title
</span>
`;
